File: Java/test/ArrayListTester.java
import Lists.ArrayList;
import Lists.LinkedList;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

public class ArrayListTester extends ListTester {
    public ArrayListTester() {
        super(new ArrayList<Integer>(Integer.class));
    }

    @Test
    public void testArrayResize() {
        final int initCapacity = getArrayList().InitCapacity;
        assertEquals(initCapacity, getArrayList().capacity());

        getArrayList().add(1);
        getArrayList().add(2);
        getArrayList().add(3);
        getArrayList().add(4);
        assertEquals(initCapacity, getArrayList().capacity());

        getArrayList().add(5);
        assertEquals(initCapacity * 2, getArrayList().capacity());
    }

    private ArrayList<Integer> getArrayList() {
        return (ArrayList<Integer>) testList;
    }
}

File: CSharp/DataStructuresTester/ArrayListTester.cs
using DataStructures.Lists;
using DataStructuresTester.Fixtures;
using Xunit;

namespace DataStructuresTester
{
    public class ArrayListTester : ListTester, IClassFixture<ArrayListFixture<int?>>
    {
        private ArrayList<int?> TestArrayList => (ArrayList<int?>)Fixture.TestList;

        public ArrayListTester(ArrayListFixture<int?> listFixture)
            : base(listFixture)
        {

        }

        [Fact]
        public void TestArrayResize()
        {
            const int initCapacity = ArrayList<int?>.InitCapacity;
            Assert.Equal(initCapacity, TestArrayList.Capacity);

            TestArrayList.Add(1);
            TestArrayList.Add(2);
            TestArrayList.Add(3);
            TestArrayList.Add(4);
            Assert.Equal(initCapacity, TestArrayList.Capacity);

            TestArrayList.Add(5);
            Assert.Equal(initCapacity * 2, TestArrayList.Capacity);
        }

        [Fact]
        public void TestToArray()
        {
            PopulateTestList();
            int?[] copyArray = TestArrayList.ToArray();

            Assert.Equal(copyArray.Length, TestArrayList.Count);
            for (int i = 0; i < TestArrayList.Count; i++)
            {
                Assert.Equal(TestArrayList[i], copyArray[i]);
            }
        }
    }
}
